src/main/java/com/minecraftabnormals/abnormals_core/core/util/DataUtil.java

```diff
@@ -314,4 +330,93 @@ public static void registerConfigPredicate(IConfigPredicateSerializer<?> seriali
+			this(modId, item, condition, behavior, (id1, id2) -> 0);
+		}
+
+		/**
+		 * Initialises a new {@link AlternativeDispenseBehavior}, where {@code condition} decides whether {@code behavior}
+		 * should be used instead of the behavior previously stored in the dispenser registry for {@code item}.
+		 *
+		 * <p>Ideally, the condition should be implemented such that the predicate only passes if the new behavior will
+		 * be 'successful', avoiding problems with failure sounds not playing.</p>
+		 *
+		 * <p>If multiple mods add a behavior to the same item and the conditions overlap such that the order that they
+		 * are registered in matters, {@code modIdComparator} (where the first parameter is {@code modId} and the second
+		 * parameter is the mod ID of another {@link AlternativeDispenseBehavior} instance)
+		 * can be used to ensure this order regardless of which mod is loaded first.</p>
+		 *
+		 * <p>For example, if a mod with the ID {@code a} has a behavior where its condition passes if any block is in front
+		 * of the dispenser, but a mod with the ID {@code b} has a behavior for the same item that passes only if a specific
+		 * block is in front of the dispenser, authors may want to make sure that {@code b}'s condition is registered after
+		 * {@code a}'s. In this case, {@code a}'s {@code modComparator} should be something like
+		 * {@code (id1, id2) -> id2.equals("b") ? -1 : 0}, and {@code b}'s should be {@code (id1, id2) -> id2.equals("a") ? 1 : 0}.</p>
+		 *
+		 * @param modId the ID of the mod registering the condition.
+		 * @param item The {@link Item} to register the {@code behavior} for.
+		 * @param condition A {@link BiPredicate} that takes in {@link IBlockSource} and {@link ItemStack} arguments,
+		 *                  returning true if {@code behavior} should be performed.
+		 * @param behavior The {@link IDispenseItemBehavior} that will be used if the {@code condition} is met.
+		 * @param modIdComparator A {@link Comparator} that compares two strings. The first is {@code modId}, and the
+		 *                        second is the mod id for another behavior registered to the same item.
+		 *                        It should return 1 if {@code behavior} is to be registered after the other behavior, -1 if
+		 *                        it should go before, and 0 in any other case.
+		 *
+		 * */
+		public AlternativeDispenseBehavior(String modId, Item item, BiPredicate<IBlockSource, ItemStack> condition, IDispenseItemBehavior behavior, Comparator<String> modIdComparator) {
+			this.modId = modId;
+			this.item = item;
+			this.condition = condition;
+			this.behavior = behavior;
+			this.modIdComparator = modIdComparator;
+		}
+
+		@Override
+		public int compareTo(AlternativeDispenseBehavior behavior) {
+			return this.item == behavior.item ? modIdComparator.compare(this.modId, behavior.modId) : 0;
+		}
+
+		public void register() {
```

Comment:
It would be good to document this method.

Reply:
Done.